### DISCUSSION TITLE
Wayland support Debian Bookworm

### DISCUSSION THREAD

**Discussion by neilbags** [regolith-linux/regolith-desktop#991] [General]
Hi, I'm trying to install regolith with sway on bookworm. 

The stable repo has a `regolith-session-sway` package but no `sway-regolith` or `xdg-desktop-portal-regolith-wayland-config` so is not installable.

I also tried debian-testing and it is broken because testing includes `libwlroots12` but regolith requires `libwlroots11`.

Did this work previously and now require rebuilding? Or is there something missing for Debian support?

**Comment by kgilmer** [comment]
Hi @neilbags , 

  Sorry I think at one point we had everything building in bookworm but then some big updates created some versioning problems for us.  I opted to remove support in bookworm, simply due to lack of time to do the backporting and testing work.  
  It may work out for you to install from the `testing` repo, however.  Note that this may also bring things in you don't want or require you to also take general changes from `testing` as a result of the change, so I suggest taking the time to read the things `apt` will pull in when you do the package install.  Here is the `apt` config you'll need to set, and I suggest replacing what you had before:

```
echo deb "[arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] \
https://regolith-desktop.org/testing-debian-testing-amd64 testing main" | \
sudo tee /etc/apt/sources.list.d/regolith.list
```
~ https://regolith-desktop.com/docs/using-regolith/install/

**Comment by alazyworkaholic** [comment]
I tried the testing packages (swapping amd64 for arm64, on an rpi3B+) and installed regolith-desktop and regolith-session-sway without errors. I changed my greeter to lightdm-gtk-greeter, which offered a choice of Regolith on Wayland or Sway Regolith. Neither worked; I was kicked back to the greeter immediately after logging in. I tried entering 'sway' from tty but command not found. Is there anything I can do to make this work?

**Reply by kgilmer** [reply]
Hi, can you try this greeter instead?

```
~$ apt list --installed | grep greeter
slick-greeter/testing,now 2.0.1-1 amd64 [installed,automatic]
```

(What a clever user alias!)

**Reply by alazyworkaholic** [reply]
I set greeter-session=slick-greeter in /etc/lightdm/lightdm.conf then rebooted. I picked the same two choices from the new greeter but was immediately kicked back the same way.
Is there any information / logs I might provide you?

**Reply by kgilmer** [reply]
I dusted of my rpi 3, [installed `arm64` `bookworm`](https://downloads.raspberrypi.com/raspios_arm64/images/raspios_arm64-2024-03-15/2024-03-15-raspios-bookworm-arm64.img.xz) and spent some time troubleshooting.  I was unable to get lightdm to present a session chooser.  I am guessing there is some lightdm configuration to enable this on the rpi but I didn't look into it.  I next tried `gdm3` and had similar behavior that you mention @alazyworkaholic .  I checked the log via `journalctl` and this seems germane:

```
Mar 30 19:00:20 raspberrypi regolith-wayland.desktop[1814]: Regolith is launching Sway with /etc/regolith/sway/config
Mar 30 19:00:21 raspberrypi regolith-wayland.desktop[2321]: 00:00:00.113 [ERROR] [wlr] [EGL] command: eglQueryDeviceStringEXT, error: EGL_BAD_PARAMETER (0x300c), message: "eglQueryDeviceStringEXT"
Mar 30 19:00:21 raspberrypi regolith-wayland.desktop[2321]: 00:00:00.113 [ERROR] [wlr] [EGL] command: eglQueryDeviceStringEXT, error: EGL_BAD_PARAMETER (0x300c), message: "eglQueryDeviceStringEXT"
Mar 30 19:00:21 raspberrypi regolith-wayland.desktop[2321]: sway-regolith: types/wlr_linux_dmabuf_v1.c:1052: wlr_linux_dmabuf_v1_create: Assertion `version <= LINUX_DMABUF_VERSION' failed.
Mar 30 19:00:21 raspberrypi regolith-wayland.desktop[1814]: /usr/bin/regolith-session-wayland: line 55:  2321 Aborted                 sway-regolith -c "$SWAY_CONFIG_FILE"
```

I am guessing there is a compat issue with sway and `armbian`.  I didn't find any promising leads, but maybe others might have an idea.

> Is there any information / logs I might provide you?

After a failed login, I typically open a tty and run `journalctl` looking at lines w/ `regolith` in them and some bad words like "failed" "error" "abort"..

**Comment by MrWazaby** [comment]
Hello, it seems that this issue is still ongoing. I cannot install sway with Regolith on Debian 12.
Do we still need to use the testing repo?

**Reply by kgilmer** [reply]
Hi @MrWazaby , due to the [older version of rustc](https://packages.debian.org/stable/rust/rustc) packaged in `bookworm`, we opted to focus on newer releases.  It's simply too costly to maintain multiple versions of each component.

From 3.2, the sway session will be supported on Debian `testing` and `trixie`, and Ubuntu `noble` when released.  If you want to use sway with bookworm, you can run it with Regolith 3.1, but it's not going to be receiving updates/fixes/etc..

**Reply by MrWazaby** [reply]
Ok thanks for the information! I will stick to X11 until the Debian 13 release it's not an issue. 

Maybe it is worth it to add this information into the documentation?

**Reply by kgilmer** [reply]
Great idea! I've added it to [3.2 release notes](https://github.com/regolith-linux/regolith-desktop-website/commit/c31a71dbbcabb65396ed37cf7b66c1cec1e14b28).  LMK if you think it might be worth putting elsewhere on the site.
